engine/storage/snapshot/src/main/java/org/apache/cloudstack/storage/snapshot/DefaultSnapshotStrategy.java

```diff
@@ -247,7 +248,7 @@ protected boolean deleteSnapshotChain(SnapshotInfo snapshot, String storageToStr
                 }
 
                 snapshot = parent;
-            }
+            } while (snapshot != null && snapshotStatesAbleToDeleteSnapshot.contains(snapshot.getState()));
```

Comment:
will this have any effect on the KVM and VMware snapshots ? @JoaoJandre

Reply:
Don't think VMware and KVM should be affected as we don't have linked snapshots there

Reply:
@harikrishna-patnala @shwstppr I actually caught this problem when working in #9270, as the same behavior of deleting the whole chain was happening there too. I have tested the fix both on main and with #9270, and it worked for both Xen and KVM (with and without incremental snaps). I currently do not have an available VMware env to test, but as @shwstppr said, VMware only works with full snapshot at the moment so this should work without any issues.